### PR TITLE
Fix bash syntax is docs deploy tag script

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit fixes some bash syntax issues in the docs deploy tag script.
These were preventing the script from running successfully as the version
number was not being constructed correctly and the script would error
when calling sphinx-build. This should fix the jobs so we no longer need
to manually upload stable documentation at release time.

### CODE REVIEW

**Review by SooluThomas — DISMISSED** [review]
LGTM! Thank you for fixing this

File: tools/deploy_documentation_tag.sh [hunk 1]
```diff
@@ -26,12 +26,12 @@ pwd
 CURRENT_TAG=`git describe --abbrev=0`
 IFS='.'
 read -ra VERSION <<< "$CURRENT_TAG"
-STABLE_VERSION="${VERSION[0]}\.${VERSION[1]}"
+STABLE_VERSION=${VERSION[0]}.${VERSION[1]}
```
Comment (SooluThomas) [inline tools/deploy_documentation_tag.sh]: I get stable version as `0.xx.0.` when I try this line locally. 

This is what I used for building the [translated ML docs](https://github.com/qiskit-community/qiskit-translations/blob/master/tools/deploy_ML_translated_documentation.sh#L23)
```suggestion
STABLE_VERSION =`echo $VERSION | cut -d "." -f -2`
```

**Review by SooluThomas — APPROVED** [review]

**Comment by jakelishman** [comment]
I don't imagine it matters much in these cases here, but the `` `command` `` syntax in bash is semi-deprecated in favour of the POSIX-compliant `$(command)`-style substitution.

**Comment by mtreinish** [comment]
Yeah, I don't think it's not a big issue for this. If ubuntu ships a bash version in the future where that syntax actively becomes a problem we can fix it then. But I expect a bunch of our scripts will have the same issue so it'll be a big thing we'll have to fix when the CI VM images update

**Pull request merged**